File: sdks/go/pkg/models/operations/postcomments.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PostCommentsQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
}

type PostCommentsRequestBodyItemTypeEnum string

const (
	PostCommentsRequestBodyItemTypeEnumFile    PostCommentsRequestBodyItemTypeEnum = "file"
	PostCommentsRequestBodyItemTypeEnumComment PostCommentsRequestBodyItemTypeEnum = "comment"
)

// PostCommentsRequestBodyItem
// The item to attach the comment to.
type PostCommentsRequestBodyItem struct {
	ID   string                              `json:"id"`
	Type PostCommentsRequestBodyItemTypeEnum `json:"type"`
}

type PostCommentsRequestBody struct {
	Item          *PostCommentsRequestBodyItem `json:"item,omitempty"`
	Message       string                       `json:"message"`
	TaggedMessage *string                      `json:"tagged_message,omitempty"`
}

type PostCommentsRequest struct {
	Retries     *utils.RetryConfig
	QueryParams PostCommentsQueryParams
	Request     *PostCommentsRequestBody `request:"mediaType=application/json"`
}

type PostCommentsResponse struct {
	ClientError *shared.ClientError
	Comment     *shared.Comment
	ContentType string
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/postcomments.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostCommentsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}

export enum PostCommentsRequestBodyItemTypeEnum {
    File = "file",
    Comment = "comment"
}


// PostCommentsRequestBodyItem
/** 
 * The item to attach the comment to.
**/
export class PostCommentsRequestBodyItem extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type: PostCommentsRequestBodyItemTypeEnum;
}


export class PostCommentsRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=item" })
  item?: PostCommentsRequestBodyItem;

  @SpeakeasyMetadata({ data: "json, name=message" })
  message: string;

  @SpeakeasyMetadata({ data: "json, name=tagged_message" })
  taggedMessage?: string;
}


export class PostCommentsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  queryParams: PostCommentsQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PostCommentsRequestBody;
}


export class PostCommentsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  comment?: shared.Comment;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
